[Mobile Payments] Add card images to Refund Confirmation screen

joshheald opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #6179 
<!-- Id number of the GitHub issue this PR addresses. -->

Will not merge until after 8.6 freezes – there's too much to tweak with the designs still.

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Adds card brand images to the Refund Via row in the Refund Confirmation screen. This is to fulfil brand guidelines from card networks, and to make it easier to identify the card in question when performing a refund.

Card images were taken from the `Hardware` target and are not final – e.g. we do not have an `Interac` image, and the `Visa` image is outdated (it should not have a gold flash any more.)

Apologies for the length - it's mostly SVGs

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

1. In the orders tab, open an order which was paid for with a Card Present payment
2. Tap `Issue Refund`
3. Select some of the order to refund, and tap `Next`
4. Observe that the card image is shown alongside the card details
5. With VoiceOver on, hear that the payment method cell is read out as `Visa card ending four two four two`

### Screenshots
<!-- Include before and after images or gifs when appropriate. -->
![Card image on refund confirmation](https://user-images.githubusercontent.com/2472348/154724225-85db247e-2493-4c77-bc3d-0f9a856ae518.png)

![card-brand-screenshot-interac](https://user-images.githubusercontent.com/2472348/154941561-0a969f1a-1b39-4584-89a6-42500a54b119.png) | ![card-brand-screenshot-visa](https://user-images.githubusercontent.com/2472348/154941569-d9929d9a-f22e-4823-a195-405b292e157f.png) | ![card-brand-screenshot-amex](https://user-images.githubusercontent.com/2472348/154941570-525a469c-5a64-43e1-9899-a3750b4de54f.png)
![card-brand-screenshot-diners](https://user-images.githubusercontent.com/2472348/154941571-8bcd9c44-4e3b-4168-abc4-aa60cd3ee9ba.png) | ![card-brand-screenshot-discover](https://user-images.githubusercontent.com/2472348/154941573-6bd19473-cb8f-4857-8e21-7fce7efa4f9e.png) | ![card-brand-screenshot-jcb](https://user-images.githubusercontent.com/2472348/154941576-938a8a49-ac14-4e5c-93db-85ae1c39c721.png)
![card-brand-screenshot-mastercard](https://user-images.githubusercontent.com/2472348/154984673-f81545d6-b2cf-4c4a-af67-2a16a250f79f.png) | ![card-brand-screenshot-unionpay](https://user-images.githubusercontent.com/2472348/154941580-5c59d8b5-f08b-40de-bf38-bcdd18538840.png) | ![card-brand-screenshot-unknown](https://user-images.githubusercontent.com/2472348/154941582-071d70f5-69df-447c-b452-68d02f14f387.png)


### Design review questions
#### Visa 
This should not have the gold flash seen in the screenshot above any more. [The Visa brand guidelines page](https://merchantsignageap.visa.com/apac_brand_guidelines) provides SVGZ files, but I can't find a free way to turn those in to SVGs for use in the app – Adobe CC will do it I think. Please could you get the up to date logo as an SVG for me?

#### Interac
I've used the [Interac logo SVG from WikiMedia](https://commons.wikimedia.org/wiki/File:InteracLogo.svg) – I'm not sure whether we actually want to do that, but it was provided by Interac and it's in line with their [brand guidelines](https://interacposcentre.ca/images/INTERAC_Brand_Guidelines_Vendors_Apr2021_FINAL.pdf), except that it includes a ® stamp.

#### Discover and Diners club
The brand guidelines suggest that the images we have (taken from WCPay) are outdated. There are EPS files for both [available to download on discoverglobalsinage.com](https://discoverglobalsignage.com/free-signage-logos). I think the correct ones are the Acceptance Marks using RGB colour. Similar to Visa, I don't have any tools suitable for doing the conversion to SVG myself: would you be able to convert them please?

#### Amex & UnionPay
I wasn't able to find useful branding guidelines for these two card networks, but I think what we have here is appropriate. An internal Amex document suggested that the `AMEX` square should only be used at very small sizes, and I think that the size we have here is appropriate.

#### Brand guidelines used
Where I could find them, I checked our use of these marks against the brand guidelines that are available.

- [Discover and Diners Club](https://discoverglobalsignage.com/uploads/09-12-21_03:59_DGN_BG_10052021_AccMk.pdf)
- [Mastercard](https://brand.mastercard.com/brandcenter/branding-requirements/mastercard.html)
- [Visa](https://spponeimages.azureedge.net/prod/a8c3df05-c491-404b-b41b-e6bfe0d77e3b2a_VDB-experience-guidelines-2018.pdf)
- [JCB](https://www.global.jcb/en/about-us/brand-concept/agreement/index.html)


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

joshheald committed:
6179 Add card images to Refund Confirmation screen

joshheald committed:
6179 Add Interac logo to refund card details

joshheald committed:
6179 Update unit test properties and card details

joshheald committed:
6179 Test for card icons

joshheald commented on Hardware/Hardware/CardReader/CardBrand.swift:
```diff
@@ -18,6 +18,8 @@ public enum CardBrand: String, CaseIterable, Codable {
     /// Diners Club card
     case dinersClub
 
+    //TODO: Add Interac here
```
Will do this in a separate PR

joe-keenan commented:
Hi @joshheald!

What you have looks pretty good, just thought I’d tweak the size so the cards are a little bit smaller, so they line up with the height of the two lines of text more neatly, and so that there’s an even amount of space around all sides of them.

I’ve also rebuilt the Discover and Visa images and used an online converter to do the Diner’s Club one (I’ve only just requested Adobe – haven’t had need of it until now!). You should be able to export SVGs directly from Figma.

I’ve mocked up a few examples, with redlines, and put the new images here: https://www.figma.com/file/Y3QRKOQJjKDdcmjxrSslpRyB/Orders---Refunds-%26-Cancelations?node-id=3854%3A6867

Let me know if you need anything else!

koke commented:
> [The Visa brand guidelines page](https://merchantsignageap.visa.com/apac_brand_guidelines) provides SVGZ files, but I can't find a free way to turn those in to SVGs for use in the app

For future reference, it seems SVGZ is just gzipped SVG. You can rename the files to `.svg.gz`, double click, and Finder will extract them to SVG.

koke commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:
```diff
@@ -322,4 +354,35 @@ private extension WCPayCardBrand {
+        case .discover:
+            return "Discover"
+        case .interac:
+            return "interac"
```
```suggestion
            return "Interac"
```

joshheald replied:
Good spot

joe-keenan commented:
Thanks @koke! That’s really handy to know.

koke commented on WooCommerce/Classes/ViewRelated/CardPresentPayments/WCPayCardBrand+icons.swift:
```diff
@@ -0,0 +1,47 @@
+
+    /// Icon name to be used for the card icon UIImage
+    var iconName: String {
+        switch self {
```
This is fine, but I wonder if it's worth simplifying here and returning `"card-brand-\(rawValue)"` instead

joshheald replied:
I used the switch so that we'd provide a placeholder image even if we add a new card brand without providing an image for it in future... so I think I'll leave this one if that's OK 😊

koke commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:
```diff
@@ -302,6 +314,24 @@ private extension WCPayCardBrand {
+
+        let brandNameNSRange = NSRange(brandNameRange, in: localizedDescription)
+        let last4NSRange = NSRange(last4Range, in: localizedDescription)
+        attributedLocalizedDescription.setAttributes([.accessibilitySpeechLanguage: "en-US"], range: brandNameNSRange)
```
I though that was a nice touch until I actually tested it. It sounds off when it uses a different voice and pronunciation. While the card brands are mostly English words, they're often adapted (mispronounced?) to the local language, so it would sound off with American English. Forcing `en-US` might even sound odd in British/Canadian/Australian/... settings.

joshheald replied:
https://user-images.githubusercontent.com/2472348/155321169-20504fbe-9da5-489f-9698-14d2a43a11b8.mp4

Agreed, it sounds better without. I'll remove this bit.

koke reviewed (approved):
Left some minor comments but the implementation is looking pretty good to me

joshheald commented:
> Hi @joshheald!
> 
> What you have looks pretty good, just thought I’d tweak the size so the cards are a little bit smaller, so they line up with the height of the two lines of text more neatly, and so that there’s an even amount of space around all sides of them.
> 
> I’ve also rebuilt the Discover and Visa images and used an online converter to do the Diner’s Club one (I’ve only just requested Adobe – haven’t had need of it until now!). You should be able to export SVGs directly from Figma.
> 
> I’ve mocked up a few examples, with redlines, and put the new images here: https://www.figma.com/file/Y3QRKOQJjKDdcmjxrSslpRyB/Orders---Refunds-%26-Cancelations?node-id=3854%3A6867
> 
> Let me know if you need anything else!

Here's the screenshots with the three updated cards

Visa | Discover | Diners
---|---|---
![Visa](https://user-images.githubusercontent.com/2472348/155172634-09f97160-887e-425c-9cae-148b931cfd86.png) | ![Discover](https://user-images.githubusercontent.com/2472348/155172646-295db86c-d568-47ae-91bf-13814402b37f.png) | ![Diners](https://user-images.githubusercontent.com/2472348/155173773-026b1675-0e30-4838-beaf-6b4534a10756.png)


Diners looks a little distorted at these small sizes unfortunately.

joshheald committed:
6179 Replace Visa, Discover, and Diners Club icons

joshheald commented:
https://user-images.githubusercontent.com/2472348/155178624-a669ab50-220f-4175-ab98-cbaa5235170f.mp4

@koke I got the voiceover localization to work

joshheald committed:
6179 Case correction in Imageset

The tests for card images were failing on CI _even though they passed locally and showed up in the app_. I suspect this is due to CI having a case-sensitive filesystem.

joe-keenan commented:
Yeah, that’s not great. I liked their earlier one with just the logo much better.
The others are looking acceptable, though. I think the padding is good now.
